x/xvalidator/validator.go

```diff
@@ -0,0 +1,29 @@
+package xvalidator
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/mesg-foundation/core/x/xnet"
+	validator "gopkg.in/go-playground/validator.v9"
+)
+
+// IsDomainName validates if given field is valid domain name.
+func IsDomainName(fl validator.FieldLevel) bool {
+	if fl.Field().String() == "" {
+		return true
+	}
+	return xnet.IsDomainName(fl.Field().String())
```

Comment:
@krhubert why using a custom domain name verification?

This https://godoc.org/gopkg.in/go-playground/validator.v9#hdr-Hostname_RFC_952 or https://godoc.org/gopkg.in/go-playground/validator.v9#hdr-Hostname_RFC_1123 are not enough?

Reply:
Hostname != domainname

https://github.com/golang/go/blob/master/src/net/dnsclient.go#L63

// See RFC 1035, RFC 3696.